jetbrains-core/src/software/aws/toolkits/jetbrains/core/AwsResourceCache.kt

```diff
@@ -50,6 +55,45 @@ interface AwsResourceCache {
         forceFetch: Boolean = false
     ): CompletionStage<T>
 
+    /**
+     * Blocking version of [getResource]
+     *
+     * @param[useStale] if an exception occurs attempting to refresh the resource return a cached version if it exists (even if it's expired). Default: true
+     * @param[forceFetch] force the resource to refresh (and update cache) even if a valid cache version exists. Default: false
+     */
+    fun <T> getResourceNow(resource: Resource<T>, timeout: Duration = DEFAULT_TIMEOUT, useStale: Boolean = true, forceFetch: Boolean = false): T =
+        wait(timeout) { getResource(resource, useStale, forceFetch) }
+
+    /**
+     * Blocking version of [getResource]
+     *
+     * @param[region] the specific [AwsRegion] to use for this resource
+     * @param[credentialProvider] the specific [ToolkitCredentialsProvider] to use for this resource
+     */
+    fun <T> getResourceNow(
+        resource: Resource<T>,
+        region: AwsRegion,
+        credentialProvider: ToolkitCredentialsProvider,
+        timeout: Duration = DEFAULT_TIMEOUT,
+        useStale: Boolean = true,
+        forceFetch: Boolean = false
+    ): T = wait(timeout) { getResource(resource, region, credentialProvider, useStale, forceFetch) }
+
+    /**
+     * Gets the [resource] if it exists in the cache.
+     *
+     * @param[useStale] if an exception occurs attempting to refresh the resource return a cached version if it exists (even if it's expired). Default: true
```

Comment:
Statement about the exception doesnt seem to match impl

Reply:
will fix